vignettes/execution.Rmd

```diff
@@ -0,0 +1,97 @@
+
+## `axecute()`
+The easiest of the execution methods to use is `axecute()`.  This function can be used to exeucte code from an R terminal or using command line scripts. A log is 
+set-up around the program, and its code is run safely and loudly (using `safely()` from the `purrr` package).
+```{r axecute, eval = FALSE}
+axecute("my_script.R")
+```
+
+## `log_*()` functions
+The `log_*()` family of functions are the core of ```logrx``` and are used internally to configure, write, and remove the log and other objects.  While these are exported functions of the package, they are intended to be used in special cases where `axecute()` will not work.
+
+Use the `log_*()` functions:
+
+* `log_init()` to create the environment log.rx
+
+* `log_config()` to add the core elements of the log to the environment, and basic elements that are available at the time of configuration
+
+* `run_safely_loudly()` to execute the program code, and capture errors, warnings,
+messages, output, and result.  This must be passed an executable R file to run and create the log elements from.
+
+* `log_write()` to generate and format the log
+
+* `log_remove()` to remove the log.rx environment created by code exeuction
+
+```{r log_*, eval = FALSE}
+log_config("my_script.R")
+run_safely_loudly("my_script.R")
+log_write()
+log_remove()
+```
+
+## Command Line Execution
+While exeucting from an R terminal is nice if you have access to one, you can also execute your code using system command line.  This is done using the `Rscript -e` command which executes a file using the registered Rscript executable.  Below are a few examples of how to use the command line to execute a file and create a log as well as how to manipulate the outputs of the execution.  These are likely to be advanced examples for most users.
+
+The below chunk will run the file my_script.R and output any standard output that is created by the execution of the file to the default location.
+```{r, engine = 'bash', eval = FALSE}
+Rscript -e "logrx::axecute('my_script.R', log_name = 'my_script.log', log_path = '.')"
+```
+
+The below chunk will run the file my_script.R and output any standard output that is created by the execution of the file to a file called temp.log.  This output can then later be accessed should the user want to check the output of the file execution.
+```{r, engine = 'bash', eval = FALSE}
+Rscript -e "logrx::axecute('my_script.R', log_name = 'my_script.log', log_path = '.')" > temp.log 2>&1
+```
+
+The below chunk will run the file my_script.R and output any standard output that is created by the execution of the file to a file called temp.log.  It will then take the contents of the temp.log file and add those to the end of the log file generated by the file execution and remove the temporary output log.
```

Comment:
```suggestion
The below chunk will run the file `my_script.R` and output any standard output that is created by the execution of the file to a file called `temp.log`.  It will then take the contents of the temp.log file and add those to the end of the log file generated by the file execution and remove the temporary output log.
```